src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java

```diff
@@ -204,7 +205,7 @@ public OffenderSummary recallOffenderBooking(@Valid final RecallBooking recallBo
 
     @Override
     public Movement getMovementByBookingIdAndSequence(final Long bookingId, final Integer sequenceNumber) {
-        return movementsService.getMovementByBookingIdAndSequence(bookingId, sequenceNumber);
+        return movementsService.getMovementByBookingIdAndSequence(bookingId, sequenceNumber).orElseThrow(EntityNotFoundException.withMessage(format("Movement Not found booking Id %d, seq %d", bookingId, sequenceNumber)));
```

Comment:
Why did you have to change the service to Optional?

Reply:
I did it to make it more consistent with custody api code and also realised there would be a null pointer if no movement was found.